docs/source/examples/e2e_flow.rst

```diff
@@ -0,0 +1,519 @@
+        # this should match the output_dir specified during
+        # finetuning
+        checkpoint_dir: <checkpoint_dir>
+
+        # checkpoint files for the fine-tuned model. This should
+        # match what's shown in the logs above
+        checkpoint_files: [
+            hf_model_0001_0.pt,
+            hf_model_0002_0.pt,
+        ]
+
+        output_dir: <checkpoint_dir>
+        model_type: LLAMA2
+
+    # Make sure to update the tokenizer path to the right
+    # checkpoint directory as well
+    tokenizer:
+        _component_: torchtune.models.llama2.llama2_tokenizer
+        path: <checkpoint_dir>/tokenizer.model
+
+
+Now, let's run the recipe.
+
+.. code-block:: bash
+
+    tune run eleuther_eval --config ./custom_eval_config.yaml
+
+
+The results should look something like this.
+
+.. code-block:: bash
+
+    [evaluator.py:324] Running loglikelihood requests
+    [eleuther_eval.py:195] Eval completed in 121.27 seconds.
+    [eleuther_eval.py:197] truthfulqa_mc2: {'acc,none': 0.489 ...
+
+So seems like our fine-tuned model gets ~48% on this task, which is ~10 points
+better than the baseline. Great! Seems like our fine-tuning helped.
+
+|
+
+Generation
+-----------
+
+We've run some evaluations and the model seems to be doing well. But does it really
+generate meaningful text for the prompts you care about? Let's find out!
+
+For this, we'll use the
+`generate recipe <https://github.com/pytorch/torchtune/blob/main/recipes/generate.py>`_
+and the associated
+`config <https://github.com/pytorch/torchtune/blob/main/recipes/configs/generation.yaml>`_.
+
+
+Let's first copy over the config to our local working directory so we can make changes.
+
+.. code-block:: bash
+
+    tune cp generation ./custom_generation_config.yaml
+
+Let's modify ``custom_generation_config.yaml`` to include the following changes.
+
+.. code-block:: yaml
+
+    checkpointer:
+        _component_: torchtune.utils.FullModelHFCheckpointer
+
+        # directory with the checkpoint files
+        # this should match the output_dir specified during
+        # finetuning
+        checkpoint_dir: <checkpoint_dir>
+
+        # checkpoint files for the fine-tuned model. This should
+        # match what's shown in the logs above
+        checkpoint_files: [
+            hf_model_0001_0.pt,
+            hf_model_0002_0.pt,
+        ]
+
+        output_dir: <checkpoint_dir>
+        model_type: LLAMA2
+
+    # Make sure to update the tokenizer path to the right
+    # checkpoint directory as well
+    tokenizer:
+        _component_: torchtune.models.llama2.llama2_tokenizer
+        path: <checkpoint_dir>/tokenizer.model
+
+
+Once the config is updated, let's kick off generation! We'll use the
+default settings for sampling with ``top_k=`300`` and a
+``temperature=0.8``. These parameters control how the probabilities for
+sampling are computed. ``top_k`` restricts the space of samples to the tokens
+with the top k probabilities.
```

Comment:
If you're gonna explain top k, probably do the same for temperature

Reply:
Let me remove this, I dont think I want to add this info here.